### ISSUE TITLE
InlineModelAdmin

### ISSUE THREAD

**Issue by Luperi** [rkhleics/wagtailmodeladmin#44]
From [Django documentation](https://docs.djangoproject.com/es/1.9/ref/contrib/admin/#inlinemodeladmin-objects), "inlines" does not work. I need the possibility to edit multiple models in a single view.

**Comment by ababic** [comment]
Hi @Luperi. `InlineModelAdmin` is a specific implementation for django's default admin area. wagtailmodeladmin, although similar in logic, is a different thing entirely, designed to get things to work in Wagtail's CMS area. Wagtail has it's own solution for inline editing, in the form of the `InlinePanel`, but it seems they'll only work on models that extend from `modelcluster.models.ClusterModel`

Basically, make your model extend from `ClusterModel`, then add a `panels` attribute to it, which should be a list/tuple of the various edit_panels provided by Wagtail: http://docs.wagtail.io/en/v1.3.1/advanced_topics/customisation/page_editing_interface.html

**Comment by Luperi** [comment]
Please, can you give me a short example? Thank you soo much.

**Comment by Luperi** [comment]
@ababic thank you for you reply, but I don't understand what I have to do. It works only if my model extends from `Page`, not from `ClusterableModel`. How can I use the `panels` attribute? Can you please give me a short example? Than you very much.

**Comment by ababic** [comment]
@Luperi You have a few choices as to what your model extends. If it's a 'Page type' you're creating, and objects need to live as pages as part of the overall tree structure, then you should extend `wagtail.wagtailcore.models.Page`. If not, you can extend whatever model class you like.

Defining panels for a page type is a little different to defining them on other models. For pages, wagtail gives you a tabbed page editing interface, and you define separate lists of panels for each one. This is all covered in the wagtail documentation here:
http://docs.wagtail.io/en/v1.3.1/topics/pages.html

And more in-depth information about edit panels here:
http://docs.wagtail.io/en/v1.3.1/advanced_topics/customisation/page_editing_interface.html

Wagtail's `Page` model extends `modelcluster.models.ClusterModel` already, so `InlinePanel` will work automatically for custom page types.

**Comment by ababic** [comment]
@Luperi If you're just creating a simple model that isn't a page, you use a single `panels` attribute instead, like in this example:

http://docs.wagtail.io/en/v1.3.1/topics/snippets.html

**Comment by Luperi** [comment]
@ababic I have a many-to-one relationship. For each `Product`, there are many `CommissionScale`.

I'm trying in this way, after have seen your examples:

```
class Product(models.Model):
    name = models.CharField(max_length = 500, null = False, blank = False)
    description = models.CharField(max_length = 500, null = True, blank = True)
    unit_of_measure = models.CharField(max_length = 50, null = False, blank = False)
    price = models.FloatField(null = False)

    create_date = models.DateTimeField(auto_now_add = True)
    mod_date = models.DateTimeField(auto_now = True)

    panels = [
        FieldPanel('name'),
        FieldPanel('description'),
        InlinePanel('commission_scale', label="Commission scales"),
    ]

    def __unicode__(self):
        return self.name

```

```
class CommissionScale(models.Model):
    fk_level = models.ForeignKey(Level, verbose_name = "level", null = True, on_delete = models.SET_NULL)
    value = models.FloatField(null = False)
    points = models.FloatField(null = False)

    def __unicode__(self):
        return self.fk_product.name + " at " + str(self.fk_level.number) + " = " + str(self.value)


```

```
class ProductCommissionScale(Orderable, CommissionScale):
    page = models.ForeignKey('demo.Product', related_name='commission_scale')

```

What's wrong??? 

Thank you for your preciuous help.

**Comment by ababic** [comment]
My advice would be to try:

#### Add these to your imports at the top:
```
from modelcluster.models import ClusterableModel
from modelcluster.fields import ParentalKey
```

#### Product should extend ClusterModel
```
class Product(ClusterModel):
    ...
    panels = [
        FieldPanel('name'),
        FieldPanel('description'),
        InlinePanel('commission_scales', label="Commission scales"),
    ]
    ...

```

#### CommissionScale should be 'abstract'
```
class CommissionScale(models.Model):
    ...
   class Meta:
       abstract = True
```

#### ProductCommissionScale should use ParentalKey
```
class ProductCommissionScale(Orderable, CommissionScale):
    product = ParentalKey('demo.Product', related_name='commission_scales')
```

**Comment by Luperi** [comment]
Ok, now it works, but why CommissionScale should be abstract in your opinion? It works if false too.

**Comment by ababic** [comment]
Because, If a CommissionScale object is only relevant in the context of a Product, making it abstract would give you the simplest database representation. If you'd rather go for multi-table inheritance, that's your call.

**Comment by Luperi** [comment]
With this extension, is there the possibility to add in wagtailadmin menu links to specific pages? 

I have to implement an advanced interface that allows my user to insert and manage orders, and to export them in a CSV file too. Maybe wagtail admin area isn't the best place to do this, is it? So I would insert a link to an external page. In your opinion, is it the best way? Thanks.

**Comment by ababic** [comment]
If you just want to add a menu item, I would suggest you take a look at the documentation on wagtail hooks:
http://docs.wagtail.io/en/v1.3.1/reference/hooks.html

If you did want to add further admin views for your model, I can't do that work for you, but you might want to look in detail at the methods on the ModelAdmin class:
https://github.com/rkhleics/wagtailmodeladmin/blob/master/wagtailmodeladmin/options.py

And perhaps check out the Treebeard recipe code, for an example of how additional functionality can be added:
https://github.com/rkhleics/wagtailmodeladmin/tree/master/wagtailmodeladmin/recipes/treebeard
